tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph/graph_container_test.ts

```diff
@@ -0,0 +1,630 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+/**
+ * Unit tests for the the graph structure component and container.
+ */
+import {CommonModule} from '@angular/common';
+import {TestBed} from '@angular/core/testing';
+import {By} from '@angular/platform-browser';
+
+import {Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+
+import {graphOpFocused} from '../../actions';
+import {DebuggerComponent} from '../../debugger_component';
+import {DebuggerContainer} from '../../debugger_container';
+import {State} from '../../store/debugger_types';
+import {
+  getFocusedGraphOpConsumers,
+  getFocusedGraphOpInfo,
+  getFocusedGraphOpInputs,
+} from '../../store';
+import {
+  createDebuggerState,
+  createState,
+  createTestGraphOpInfo,
+} from '../../testing';
+import {GraphContainer} from './graph_container';
+import {GraphModule} from './graph_module';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Graph Container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [DebuggerComponent, DebuggerContainer],
+      imports: [CommonModule, GraphModule],
+      providers: [
+        provideMockStore({
+          initialState: createState(createDebuggerState()),
+        }),
+        DebuggerContainer,
+      ],
+    }).compileComponents();
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    dispatchSpy = spyOn(store, 'dispatch');
+  });
+
+  it('renders no-op-selected element and nothing else when no op is selected', () => {
+    const fixture = TestBed.createComponent(GraphContainer);
+    store.overrideSelector(getFocusedGraphOpInfo, null);
+    fixture.detectChanges();
+
+    const noOpFocused = fixture.debugElement.query(By.css('.no-op-focused'));
+    expect(noOpFocused).not.toBeNull();
+    expect(noOpFocused.nativeElement.innerText).toMatch(/No graph op selected/);
+    const inputsContainer = fixture.debugElement.query(
+      By.css('.inputs-container')
+    );
+    expect(inputsContainer).toBeNull();
+    const selfOpContainer = fixture.debugElement.query(
+      By.css('.self-op-container')
+    );
+    expect(selfOpContainer).toBeNull();
+    const consumersContainer = fixture.debugElement.query(
+      By.css('.consumers-container')
+    );
+    expect(consumersContainer).toBeNull();
+  });
+
+  for (const inputOutputSlot of [0, 1]) {
+    for (const consumerInputSlot of [0, 1]) {
+      for (const neighborDataAvailable of [false, true]) {
+        it(
+          `renders op with 1 input tensor and 1 consumer: ` +
```

Comment:
FIX ME: this `it` will generate the same string for different `inputOutputSlot`.

Reply:
Done.